crates/ty/src/lib.rs

```diff
@@ -355,9 +354,9 @@ impl MainLoop {
 /// A progress reporter for `ty check`.
 struct IndicatifReporter(indicatif::ProgressBar);
 
-impl Default for IndicatifReporter {
-    fn default() -> IndicatifReporter {
-        let progress = indicatif::ProgressBar::new(0);
+impl ty_project::Reporter for IndicatifReporter {
+    fn with_files(files: usize) -> IndicatifReporter {
```

Comment:
The fact that all check functions now need to be generic (and need to specify the type as well) is somewhat unfortunate. 

I'd have a slight preference towards using an associated type or returning a `Box<dyn>` from `with_files` instead.